app/src/main/java/org/simple/clinic/drugs/search/DrugsSearchScreen.kt

```diff
@@ -105,6 +113,9 @@ class DrugsSearchScreen : BaseScreen<
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
 
+    val drugFrequencyChoiceItems = DrugFrequencyChoiceItems(drugFrequencyFactory.provideFields())
+    drugFrequencyToFrequencyChoiceItemMap = drugFrequencyChoiceItems.items.associateBy({ it.drugFrequency }, { it })
```

Comment:
Can't we just use this instead? 🤔 . We can even do this inside the adapter as well and just pass in the factory/items in the screen.

```suggestion
val drugFrequencyToFrequencyChoiceItemMap = drugFrequencyFactory
    .provideFields()
    .associateBy({ it.drugFrequency }, { it })
```